src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java

```diff
@@ -335,12 +337,13 @@ else if (canReuseStackmaps) {
                                                   handlers);
                     maxStack = gen.maxStack();
                     maxLocals = gen.maxLocals();
-                    if (buf.getMajorVersion() >= Classfile.JAVA_6_VERSION) {
-                        stackMapAttr = gen.stackMapTableAttribute();
-                    } else {
-                        stackMapAttr = null;
-                    }
+                    stackMapAttr = gen.stackMapTableAttribute();
                 }
+                else {
+                    maxStack = maxLocals = 255;
```

Comment:
Will this have a negative effect, such as slowing down the vm at runtime?

Reply:
I've added I've implemented StackCounter as a simple alternative to StackMapGenerator